Title: User-defined Prompt Colors
Author: Eric Pruitt (https://www.codevat.com/, https://github.com/ericpruitt/)

With this patch, the prompt foreground can be specified with the "-pf" command
line option, and the background can be specified with "-pb". A new entry must
be added to the color definition array in the configuration header after
applying this patch:

    static const char *colors[SchemeLast][2] = {
        // ...
        [SchemePrompt] = { "#000000", "#ffffff" },
    };

diff --git a/dmenu.c b/dmenu.c
index df51e76..dea1700 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -25,7 +25,7 @@
 #define TEXTW(X)              (drw_fontset_getwidth(drw, (X)) + lrpad)
 
 /* enums */
-enum { SchemeNorm, SchemeSel, SchemeOut, SchemeLast }; /* color schemes */
+enum { SchemeNorm, SchemeSel, SchemeOut, SchemePrompt, SchemeLast }; /* color schemes */
 
 struct item {
 	char *text;
@@ -136,7 +136,7 @@ drawmenu(void)
 	drw_rect(drw, 0, 0, mw, mh, 1, 1);
 
 	if (prompt && *prompt) {
-		drw_setscheme(drw, scheme[SchemeSel]);
+		drw_setscheme(drw, scheme[SchemePrompt]);
 		x = drw_text(drw, x, 0, promptw, bh, lrpad / 2, prompt, 0);
 	}
 	/* draw input field */
@@ -530,6 +530,7 @@ setup(void)
 	scheme[SchemeNorm] = drw_scm_create(drw, colors[SchemeNorm], 2);
 	scheme[SchemeSel] = drw_scm_create(drw, colors[SchemeSel], 2);
 	scheme[SchemeOut] = drw_scm_create(drw, colors[SchemeOut], 2);
+	scheme[SchemePrompt] = drw_scm_create(drw, colors[SchemePrompt], 2);
 
 	clip = XInternAtom(dpy, "CLIPBOARD",   False);
 	utf8 = XInternAtom(dpy, "UTF8_STRING", False);
@@ -637,6 +638,10 @@ main(int argc, char *argv[])
 			colors[SchemeNorm][ColBg] = argv[++i];
 		else if (!strcmp(argv[i], "-nf"))  /* normal foreground color */
 			colors[SchemeNorm][ColFg] = argv[++i];
+		else if (!strcmp(argv[i], "-pf"))  /* prompt foreground color */
+			colors[SchemePrompt][ColFg] = argv[++i];
+		else if (!strcmp(argv[i], "-pb"))  /* prompt background color */
+			colors[SchemePrompt][ColBg] = argv[++i];
 		else if (!strcmp(argv[i], "-sb"))  /* selected background color */
 			colors[SchemeSel][ColBg] = argv[++i];
 		else if (!strcmp(argv[i], "-sf"))  /* selected foreground color */
